sdk/core/Azure.Core/src/RequestOptions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure
+{
+    /// <summary>
+    /// Options that can be used to control the behavior of a request sent by a client.
+    /// </summary>
+    public class RequestOptions
+    {
+        private bool _frozen;
+
+        private (int Status, bool IsError)[]? _statusCodes;
+        internal (int Status, bool IsError)[]? StatusCodes => _statusCodes;
+
+        private ResponseClassificationHandler[]? _handlers;
+        internal ResponseClassificationHandler[]? Handlers => _handlers;
+
+        internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestOptions"/> class.
+        /// </summary>
+        public RequestOptions()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestOptions"/> class
+        /// that is a copy of the passed-in options.
+        /// <paramref name="options">The RequestOptions to copy.</paramref>
+        /// </summary>
+        protected RequestOptions(RequestOptions options)
```

Comment:
Should we consider decorating with `[EditorBrowsable(EditorBrowsableState.Never)]` to remove this from docs and code completion?